### DISCUSSION TITLE
Personal Folders not being created

### DISCUSSION THREAD

**Discussion by Esslingers** [nilsteampassnet/TeamPass#4246] [Q&A]
I have a new install of Ubuntu 24 and using git to pull the code for Teampass version 3.1.2.45.  I have each user check box set to allow personal folders and I have task set to "Ensure users personal folder are created", yet personal folders are never created.  How can I manually create the personal folders? Is there some setup that I'm missing for this automation to work?

**Comment by Mwendlin** [comment]
i had the same problem, after some trial and error i realized that there are some typos in the scripts to be executed which is why the whole thing doesn't work.
try this in your task_maintenance_users_personal_folder.php : 


<?php
/**
 * Teampass - a collaborative passwords manager.
 * ---
 * This file is part of the TeamPass project.
 * 
 * TeamPass is free software: you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by
 * the Free Software Foundation, version 3 of the License.
 * 
 * TeamPass is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 * 
 * Certain components of this file may be under different licenses. For
 * details, see the `licenses` directory or individual file headers.
 * ---
 * @file      task_maintenance_users_personal_folder.php
 * @author    Nils Laumaillé (nils@teampass.net)
 * @copyright 2009-2024 Teampass.net
 * @license   GPL-3.0
 * @see       https://www.teampass.net
 */

use TeampassClasses\NestedTree\NestedTree;
use TeampassClasses\SessionManager\SessionManager;
use Symfony\Component\HttpFoundation\Request;
use TeampassClasses\Language\Language;
use TeampassClasses\ConfigManager\ConfigManager;

// Load functions
require_once __DIR__.'/../sources/main.functions.php';

// init
loadClasses('DB');
$lang = new Language($session->get('user-language') ?? 'english');

// Load config if $SETTINGS not defined
$configManager = new ConfigManager();
$SETTINGS = $configManager->getAllSettings();

// Define Timezone
date_default_timezone_set(isset($SETTINGS['timezone']) === true ? $SETTINGS['timezone'] : 'UTC');

// Set header properties
header('Content-type: text/html; charset=utf-8');
header('Cache-Control: no-cache, no-store, must-revalidate');
error_reporting(E_ERROR);
// increase the maximum amount of time a script is allowed to run
set_time_limit($SETTINGS['task_maximum_run_time']);

// --------------------------------- //

require_once __DIR__.'/background_tasks___functions.php';

// log start
$logID = doLog('start', 'do_maintenance - users-personal-folder', 1);

// Perform maintenance tasks
createUserPersonalFolder();

// log end
doLog('end', '', 1, $logID);

/**
 * Permits to create the personal folder for each user.
 *
 * @return void
 */
function createUserPersonalFolder(): void
{
    //Libraries call
    $tree = new NestedTree(prefixTable('nested_tree'), 'id', 'parent_id', 'title');

    //get through all users
    $users = DB::query(
        'SELECT id, login, email
        FROM ' . prefixTable('users') . '
        WHERE id NOT IN ('.OTV_USER_ID.', '.TP_USER_ID.', '.SSH_USER_ID.', '.API_USER_ID.')
        ORDER BY login ASC'
    );
    foreach ($users as $user) {
        /*
        //update PF field for user
        DB::update(
            prefixTable('users'),
            array(
                'personal_folder' => '1',
            ),
            'id = %i',
            $user['id']
        );
        */

        //if folder doesn't exist then create it
        $data = DB::queryfirstrow(
            'SELECT id
            FROM ' . prefixTable('nested_tree') . '
            WHERE title = %s AND parent_id = %i',
            $user['id'],
            0
        );
        $counter = DB::count();
        if ($counter === 0) {
            //If not exist then add it
            DB::insert(
                prefixTable('nested_tree'),
                array(
                    'parent_id' => '0',
                    'title' => $user['id'],
                    'personal_folder' => '1',
                    'categories' => '',
                )
            );

            //rebuild fuild tree folder
            $tree->rebuild();
        } else {
            //If exists then update it
            DB::update(
                prefixTable('nested_tree'),
                array(
                    'personal_folder' => '1',
                ),
                'title=%s AND parent_id=%i',
                $user['id'],
                0
            );
            //rebuild fuild tree folder
            $tree->rebuild();

            // Get an array of all folders
            $folders = $tree->getDescendants($data['id'], false, true, true);
            foreach ($folders as $folder) {
                //update PF field for user
                DB::update(
                    prefixTable('nested_tree'),
                    array(
                        'personal_folder' => '1',
                    ),
                    'id = %s',
                    $folder
                );
            }
        }

        // Ensure only the user items have a sharekey
        purgeUnnecessaryKeys(false, $user['id']);
    }
}

**Comment by Esslingers** [comment]
where specifically is the typo?  your corrections came though as a single line of text instead of formatted php

**Comment by Mwendlin** [comment]
unfortunately the content can't be posted here properly which is why i tried to attach the file as .txt but that didn't work either, maybe it will work now
[task_maintenance_users_personal_folder.php.txt](https://github.com/user-attachments/files/16348771/task_maintenance_users_personal_folder.php.txt)

**Comment by DobroFenix** [comment]
The task appeared, the user had an alert, the task was completed, but the folder did not appear for the user.

**Comment by hexxamillion** [comment]
I have this same issue. I just installed on Ubuntu 24.04 and scheduled the tasks and no personal folders were created. I get no errors. Everything else installed and works very nice. Just can't get the task to work for creating the personal folders. Any recommendations? Back in the day with the older versions we had an option to create the folders upon creating the account. That worked nice and did not require scheduling a task.

**Comment by rafinhamvieira** [comment]
Any return? I have the same problem, my folders aren't created.

**Reply by hexxamillion** [reply]
I still have the same issue.

**Reply by rafinhamvieira** [reply]
I solve this running the task "Ensure users personal folder are created" and when i logged with other user, works normally.
https://github.com/user-attachments/assets/344c0ee5-6dc1-4e02-8c0c-c7785b600973

**Reply by hexxamillion** [reply]
I've tried that many times. I only have option to schedule the task not to immediately execute. Does not seem to do anything.
